Dotnet tools upgrade fails if there are none installed.

**What did you expect to happen?**
Nothing


**What actually happened?**
It tried to upgrade .net tool ' ---------------------' which doesnt exist.


**Additional details:**
Installed topgrade on a fairly clean vm supplied by ms: https://developer.microsoft.com/en-us/windows/downloads/virtual-machines/
There are no global dotnet tools installed, thats probably causing parsing of the output to go haywire.

<details><pre>
error NU1101: Unable to find package ---------------------. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages, nuget.org
Tool '---------------------' failed to update due to the following:
The tool package could not be restored.
Tool '---------------------' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
</pre></details>

--- a/src/steps/generic.rs
+++ b/src/steps/generic.rs
@@ -302,12 +302,15 @@ pub fn run_composer_update(ctx: &ExecutionContext) -> Result<()> {
 pub fn run_dotnet_upgrade(ctx: &ExecutionContext) -> Result<()> {
     let dotnet = utils::require("dotnet")?;
 
-    let output = Command::new(dotnet)
-        .args(&["tool", "list", "--global"])
-        .check_output()?;
+    let output = Command::new(dotnet).args(&["tool", "list", "--global"]).output()?;
 
+    if !output.status.success() {
+        return Err(SkipStep(format!("dotnet failed with exit code {:?}", output.status)).into());
+    }
+
+    let output = String::from_utf8(output.stdout)?;
     if !output.starts_with("Package Id") {
-        return Err(SkipStep(String::from("dotnet does not output packages")).into());
+        return Err(SkipStep(String::from("dotnet did not output packages")).into());
     }
 
     let mut packages = output.split('\n').skip(2).filter(|line| !line.is_empty()).peekable();

--- a/src/steps/generic.rs
+++ b/src/steps/generic.rs
@@ -306,6 +306,10 @@ pub fn run_dotnet_upgrade(ctx: &ExecutionContext) -> Result<()> {
         .args(&["tool", "list", "--global"])
         .check_output()?;
 
+    if !output.starts_with("Package Id") {
+        return Err(SkipStep(String::from("dotnet does not output packages")).into());
+    }
+
     let mut packages = output.split('\n').skip(2).filter(|line| !line.is_empty()).peekable();
 
     if packages.peek().is_none() {

